src/main/java/uk/ac/ebi/ampt2d/accession/service/BlockManager.java

```diff
@@ -90,8 +90,8 @@ private MonotonicRange pollNextMonotonicRange(int maxSize) {
         return monotonicRange;
     }
 
-    public boolean isAvailableSpaceLessThan(int totalAccessionsToGenerate) {
-        return availableRanges.getNumOfValuesInQueue() < totalAccessionsToGenerate;
+    public boolean hasAvailableSpace(int spaceNeeded) {
```

Comment:
`accessionsNeeded` better than `spaceNeeded` (otherwise it is not definite what the space is for)